### PR TITLE
create fuse subcommand to use weed with mount

### PR DESCRIPTION
**Pull request by danielflira**
It add fuse subcommand that allow use weed as mount type.

It's related with issues: #1323 #1674

### CODE REVIEW

**Comment by prologic** [comment]
This is _really_ nice. 😍

File: weed/command/fuse.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+		// if doesn't key and value skip
+		if len(parts) != 2 {
+			continue
+		}
+
+		// switch key keeping "weed mount" parameters
+		switch parts[0] {
+			case "filer":
+				mountOptions.filer = &parts[1]
```
Comment (chrislusf) [inline weed/command/fuse.go]: filer now supports comma separated list of filers. Any way to make it work in this mode?

File: weed/command/fuse.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+		// split just first = character
+		parts := strings.SplitN(option, "=", 2)
+
+		// if doesn't key and value skip
+		if len(parts) != 2 {
+			continue
+		}
+
+		// switch key keeping "weed mount" parameters
+		switch parts[0] {
```
Comment (chrislusf) [inline weed/command/fuse.go]: how about using a variable for `parts[1]`

File: weed/command/fuse.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+}
+
+var cmdFuse = &Command{
+	UsageLine: "fuse /mnt/mount/point -o \"filer=localhost:8888,filer.remote=/\"",
+	Short: "Allow use weed with linux's mount command",
+	Long: `Allow use weed with linux's mount command
+
+  You can use -t weed on mount command:
+  mv weed /sbin/mount.weed
+  mount -t weed fuse /mnt -o "filer=localhost:8888,filer.remote=/"
```
Comment (chrislusf) [inline weed/command/fuse.go]: change `filer.remote` to `filer.path` here and a few other places?

File: weed/command/fuse.go (hunk 1 above)
Comment (chrislusf) [reply weed/command/fuse.go:49]: Does it support adding a quoted filer list?

Comment (chrislusf) [reply weed/command/fuse.go]: Seems not an issue.  Double quoting is required for this case.  For example:
```
                     mount -t tmpfs none /mnt -o \
                       'context="system_u:object_r:tmp_t:s0:c127,c456",noexec'
```
https://man7.org/linux/man-pages/man8/mount.8.html

Comment (danielflira) [reply weed/command/fuse.go]: It can be a problem. I just split commas without take care of quotes. I going to check it.

Comment (chrislusf) [reply weed/command/fuse.go]: If it works, please send a new PR to add the example to the help message.

I am going to merge this PR for now.

**Comment by chrislusf** [comment]
could you please also add a section to https://github.com/chrislusf/seaweedfs/wiki/FUSE-Mount ?

**Comment by danielflira** [comment]
> could you please also add a section to https://github.com/chrislusf/seaweedfs/wiki/FUSE-Mount ?

it was made

**Pull request merged**